tests/test_validators.py

```diff
@@ -2918,7 +2918,7 @@ class Child(BaseModel):
 
         @model_validator(mode='after')  # type: ignore
         def validate_model(self) -> 'Child':
-            return Child.model_construct(name='different!')
+            return Child.model_construct(name='name')
 
     with pytest.warns(UserWarning, match='A custom validator is returning a value other than `self`'):
-        Child(name='foo')
+        Child(name='name')
```

Comment:
I'd rather leave these as different to illustrate the odd behavior...

Reply:
To make it more clear, we could add an assert and a comment at the end (like self overrides, blah blah blah)